src/render/draw_background.js

```diff
@@ -25,7 +25,7 @@ function drawBackground(painter: Painter, sourceCache: SourceCache, layer: Backg
     const transform = painter.transform;
     const tileSize = transform.tileSize;
     const image = layer.paint.get('background-pattern');
-    if (painter.isPatternMissing(image)) return;
+    if (painter.isPatternMissing((image: any))) return;
```

Comment:
Is Flow complaining because the expected type in `isPatternMissing` is `CrossFaded<string>` but `layer.paint.get` is now returning a `CrossFaded<Image>`. If so would it make sense to update `isPatternMissing` and undo the changes in `draw_line`, `draw_fill`, and `draw_fill_extrusion`

Reply:
`isPatternMissing` expects `CrossFaded<string>`. I can change that to `CrossFaded<Image>` and then in `isPatternMissing` call `image.to.toString()` and it works well and gets rid of the Flow error.

If I try that same approach in the other `draw_*` files though, I get a weird Flow error. 

```
        if (constantPattern && tile.imageAtlas) {
            const posTo = tile.imageAtlas.patternPositions[constantPattern.to.toString()];
            const posFrom = tile.imageAtlas.patternPositions[constantPattern.from.toString()];
            if (posTo && posFrom) programConfiguration.setConstantPatternPositions(posTo, posFrom);
        }
```

gets rid of the `Image is incompatible with string` error (patternPositions expects a string key) but Flow then complains that it `Cannot get tile.imageAtlas.patternPositions because property patternPositions is missing in null or undefined` which doesn't make sense since we're already checking that `tile.imageAtlas` isn't null or undefined.

Reply:
Actually, I'm wrong. I had forgotten to remove the cast to `any` in `draw_background`. If I do that and update `isPatternMissing` I get this error:

`Cannot call painter.isPatternMissing with image bound to image because Image [1]
is incompatible with Image [2] in property from.`

which doesn't make sense since `to` and `from` are just defined as the type passed to `CrossFaded`

https://github.com/mapbox/mapbox-gl-js/blob/3d6d2b440c89199dfc38f0a2b109588f1f264ac8/src/style/properties.js#L28-L31

Reply:
@ahk helped me fix the casts to `any` in `draw_{fill/fill-extrusion/line}` but we haven't come up with a way around this cast yet. We're still trying.

Reply:
Turns out that type `Image` accidentally shadows the Javascript constructor `Image` leading to a very difficult to debug Flow error. To avoid this, I'm going to change the `Image` type to `ResolvedImage`

Reply:
All casts to `any` have been removed. The Flow typing should be solid now